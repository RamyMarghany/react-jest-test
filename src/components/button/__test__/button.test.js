import React from 'react'
import ReactDom from 'react-dom'
import Button from './../button'
import {render, cleanup} from '@testing-library/react'
import "@testing-library/jest-dom"; 
import renderer from 'react-test-renderer'

afterEach(cleanup);

it('render without crashes',()=>{
    const div = document.createElement('button');
    ReactDom.render(<Button></Button>, div)
})

it('render button correctly', ()=>{
    const {getByTestId} = render(<Button label="click me"></Button>)
    expect(getByTestId('button')).toHaveTextContent('click me')
})


it('render button correctly', ()=>{
    const {getByTestId} = render(<Button label="save"></Button>)
    expect(getByTestId('button')).toHaveTextContent('save')
})

it('matches snapshots', ()=>{
    const tree = renderer.create(<Button label='save'/>).toJSON()
    expect(tree).toMatchSnapshot();
})