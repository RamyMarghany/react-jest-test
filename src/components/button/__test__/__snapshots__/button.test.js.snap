// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshots 1`] = `
<div
  className="btn"
  data-testid="button"
>
  save
</div>
`;
